src/components/Company/PaySchedule/_parts/Edit.tsx

```
@@ -92,26 +108,50 @@ export const Edit = () => {
+                  <Select
+                    control={control}
+                    name="pay_period_preview_range"
+                    label={t('labels.preview')}
+                    items={payPeriodPreview.map((period, index) => {
+                      return {
+                        id: index,
+                        name: `${period.start_date ? new Date(period.start_date).toLocaleDateString('en-US', { weekday: 'short', month: 'short', day: 'numeric' }) : ''} – ${period.end_date ? new Date(period.end_date).toLocaleDateString('en-US', { weekday: 'short', month: 'short', day: 'numeric' }) : ''}`,
+                      }
+                    })}
+                    defaultSelectedKey={selectedPayPeriodIndex}
+                    onSelectionChange={value => {
+                      if (typeof value === 'number') {
+                        setSelectedPayPeriodIndex(value)
+                      }
+                    }}
+                  >
+                    {option => <ListBoxItem key={option.id}>{option.name}</ListBoxItem>}
+                  </Select>
+                )
               }
-              return (
-                <PayPreviewCard
-                  key={index}
-                  checkdate={new Date(payPeriod.check_date ?? '')}
-                  endDate={new Date(payPeriod.end_date ?? '')}
-                  startDate={new Date(payPeriod.start_date ?? '')}
-                  runPayrollBy={new Date(payPeriod.run_payroll_by ?? '')}
-                />
-              )
-            })}
-          {payPreviewLoading && (
-            <Flex justifyContent={'center'} alignItems={'center'}>
-              <Spinner title={t('loading')} />
-            </Flex>
+              rangeSelected={{
+                start: payPeriodPreview[selectedPayPeriodIndex].start_date || '',
+                end: payPeriodPreview[selectedPayPeriodIndex].end_date || '',
+                label: t('payPreview.payPeriod') || 'Pay Period',
```

why falling back on hardcoded string?

---

Looks like we're getting null strings back from the API here. Handled this via casting as it looks like we never get a null back for this value.